homeassistant/components/sunweg/sensor.py

```diff
@@ -153,26 +121,41 @@ def __init__(
             description.icon if description.icon is not None else "mdi:solar-power"
         )
 
-        self._attr_device_info = DeviceInfo(
-            identifiers={(DOMAIN, str(probe.plant_id))},
-            manufacturer="SunWEG",
-            name=name,
+        self._attr_device_info = coordinator.device_info
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle data update."""
+        previous_value = self.native_value
+        value: StateType | datetime.datetime = self.coordinator.get_api_value(
+            self.entity_description.api_variable_key,
+            self.device_type,
+            self.inverter_id,
+            self.deep_name,
         )
-
-    def update(self) -> None:
-        """Get the latest data from the Sun WEG API and updates the state."""
-        self.probe.update()
-        (
-            self._attr_native_value,
-            self._attr_native_unit_of_measurement,
-        ) = self.probe.get_data(
-            api_variable_key=self.entity_description.api_variable_key,
-            api_variable_unit=self.entity_description.api_variable_unit,
-            deep_name=self.deep_name,
-            device_type=self.device_type,
-            inverter_id=self.inverter_id,
-            name=self.entity_description.name,
-            native_unit_of_measurement=self.native_unit_of_measurement,
-            never_resets=self.entity_description.never_resets,
-            previous_value_drop_threshold=self.entity_description.previous_value_drop_threshold,
+        previous_unit_of_measurement: str | None = self.native_unit_of_measurement
```

Comment:
Why/how does the unit of measurement change?

Reply:
When the plant is offline, some sensors could return 0 without the actual metric. This could impact the amount of generated energy for example.